pkg/vgmanager/wipe_devices_test.go

```diff
@@ -0,0 +1,174 @@
+			removeReferenceCount: 0,
+		},
+		{
+			name:         "Both devices exist in the device list",
+			devicePaths:  []string{"/dev/loop1", "/dev/loop2"},
+			blockDevices: []lsblk.BlockDevice{{KName: "/dev/sda"}, {KName: "/dev/loop1"}, {KName: "/dev/loop2", Children: []lsblk.BlockDevice{{KName: "/dev/loop2p1"}}}},
+			nodeStatus: v1alpha1.LVMVolumeGroupNodeStatus{Spec: v1alpha1.LVMVolumeGroupNodeStatusSpec{
+				LVMVGStatus: []v1alpha1.VGStatus{
+					{Name: "vg1", Devices: []string{"/dev/sda"}}},
+			}},
+			wipeCount:            2,
+			removeReferenceCount: 1,
+		},
+		{
+			name:         "Both devices, one of them is a child, exist in the device list",
+			devicePaths:  []string{"/dev/loop1", "/dev/loop2p1"},
+			blockDevices: []lsblk.BlockDevice{{KName: "/dev/sda"}, {KName: "/dev/loop1"}, {KName: "/dev/loop2", Children: []lsblk.BlockDevice{{KName: "/dev/loop2p1"}}}},
+			nodeStatus: v1alpha1.LVMVolumeGroupNodeStatus{Spec: v1alpha1.LVMVolumeGroupNodeStatusSpec{
+				LVMVGStatus: []v1alpha1.VGStatus{
+					{Name: "vg1", Devices: []string{"/dev/sda"}}},
+			}},
+			wipeCount:            2,
+			removeReferenceCount: 0,
+		},
+		{
+			name:         "Both devices exist in the device list, one of them is part of the vg",
+			devicePaths:  []string{"/dev/loop1", "/dev/loop2"},
+			blockDevices: []lsblk.BlockDevice{{KName: "/dev/sda"}, {KName: "/dev/loop1"}, {KName: "/dev/loop2", Children: []lsblk.BlockDevice{{KName: "/dev/loop2p1"}}}},
+			nodeStatus: v1alpha1.LVMVolumeGroupNodeStatus{Spec: v1alpha1.LVMVolumeGroupNodeStatusSpec{
+				LVMVGStatus: []v1alpha1.VGStatus{
+					{Name: "vg1", Devices: []string{"/dev/loop1"}}},
+			}},
+			wipeCount:            1,
+			removeReferenceCount: 1,
+		},
+		{
+			name:         "Both devices are part of the vg",
+			devicePaths:  []string{"/dev/loop1", "/dev/loop2"},
+			blockDevices: []lsblk.BlockDevice{{KName: "/dev/sda"}, {KName: "/dev/loop1"}, {KName: "/dev/loop2", Children: []lsblk.BlockDevice{{KName: "/dev/loop2p1"}}}},
+			nodeStatus: v1alpha1.LVMVolumeGroupNodeStatus{Spec: v1alpha1.LVMVolumeGroupNodeStatusSpec{
+				LVMVGStatus: []v1alpha1.VGStatus{
+					{Name: "vg1", Devices: []string{"/dev/loop1", "/dev/loop2"}}},
+			}},
+			wipeCount:            0,
+			removeReferenceCount: 0,
+		},
+	}
+	mockWipefs := wipefsmocks.NewMockWipefs(t)
+	mockDmsetup := dmsetupmocks.NewMockDmsetup(t)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &VGReconciler{Wipefs: mockWipefs, Dmsetup: mockDmsetup}
+			if tt.wipeCount > 0 {
+				mockWipefs.EXPECT().Wipe(mock.Anything).Return(nil).Times(tt.wipeCount)
+			}
+			if tt.removeReferenceCount > 0 {
+				mockDmsetup.EXPECT().Remove(mock.Anything).Return(nil).Times(tt.removeReferenceCount)
+			}
+			volumeGroup := &v1alpha1.LVMVolumeGroup{
+				ObjectMeta: metav1.ObjectMeta{Name: "vg1"},
+				Spec: v1alpha1.LVMVolumeGroupSpec{DeviceSelector: &v1alpha1.DeviceSelector{
+					Paths:                             tt.devicePaths,
```

Comment:
These tests are missing optionalDevicePath cases I think. Did we test what happens when you pass a non existing devicePath to wipefs?

Reply:
In the implementation, I just append optionalDevicePaths to devicePaths: https://github.com/openshift/lvm-operator/pull/423/files#diff-df5f27c889c2546de41e7ad40a6dbad4e9c7933a22593c529fd23977c25b9e54R19 So, I didn't think it'd make a difference to add optionalDevicePath cases to the test. 

If the devicePath is non-existing, wipefs returns a failure like "No such file or directory". We just fail the reconciliation at that point.

Reply:
> If the devicePath is non-existing, wipefs returns a failure like "No such file or directory". We just fail the reconciliation at that point.

In a multi-node scenario, it is normal to have certain optionalDevices that are present on one node but not on another (imagine 2 devices with 2 nodes where each node has one of the devices, but not the other). In this case, it would always fail to wipe the devices on all nodes because at least one device would not be present.